File: deprecated_scripts/getmatchposition_gmap.jl
#!/usr/local/extras/Genomics/bin/julia

# BEWARE: to work properly, the gmap analysis as to be done one target only!

# parameters
fil = ARGS[1]   # gmap results
g = ARGS[2] # genome to fetch the sequence from
outpref = ARGS[3]  # prefix of output (full path but no extension)

# load gmap results all at once
f = open(fil)
txt = readlines(f)
close(f)

# check number of exons
Nex = filter(x -> contains(x, "    Number of exons:"), txt)[1]
Nex = parse(split(Nex, [ ' ', '\n'])[end-1])

# loop on all exons
for i in 1:Nex
    # get coordinates
    l = find(x -> contains(x, "  Alignment for path 1:"), txt)
    lc = l[1] + 1 + i

    # remove leading strand sign
    lt = replace(strip(txt[lc]), r"^[+-]", "")
    chr, coor = split(lt, [':', ' '])[1:2]
    sta, sto = split(strip(coor), '-')
    
    # run get-genome
    println(STDERR, "  # Fetch exon $i sequence for genome $g")
    println(STDERR, "get-genome -d $g $chr:$sta-$sto > $outpref.$i.fasta")
    run(pipeline(`get-genome -d $g $chr:$sta-$sto`, stdout="$outpref.$i.fasta"))
end

print(Nex)

File: getmatchposition_gmap.py
#!/bin/env python2

# BEWARE: to work properly, the gmap analysis as to be done on one target only!
from re import search, sub, split
from os import system
from sys import argv, stderr

# parameters
fil = argv[1]   # gmap results
g = argv[2] # genome to fetch the sequence from
outpref = argv[3]  # prefix of output (full path but no extension)

# load gmap results all at once
f = open(fil)
txt = f.readlines()
f.close()

# check number of exons
Nex = filter(lambda x:search('    Number of exons:', x), txt)[0]
Nex = int(Nex.split()[-1])

# loop on all exons
for i in range (Nex):
    # get coordinates
    l = [j for j, item in enumerate(txt) if search('  Alignment for path 1:', item)][0]
    lc = l + 1 + (i + 1)

    # remove leading strand sign
    lt = sub(r"^[+-]", "", (txt[lc]).strip())
    chr, coor = split(r"[ :]", lt)[0:2]
    sta, sto = split('-', coor.strip())[0:2]

    # run get-genome
    cmd = "get-genome -d %s %s:%s-%s > %s.%s.fasta\n" % (g, chr,
        sta, sto, outpref, i+1)
    stderr.write("  # Fetch exon %s sequence for genome %s\n" % (i+1, g))
    stderr.write(cmd)
    system(cmd)

print Nex
